connector/elastic.go

```diff
@@ -153,3 +164,61 @@ func createIndex(indexName string) (err error) {
 	}
 	return
 }
+
+func createIndexWithShardManagement(indexName string, shardsCount int, replicasCount int) (err error) {
+
+	// check if the index already exist
+	index, err := elasticClient.Search(indexName).Do(context.Background())
+	if err != nil { // index doesn't exist
+		reason := fmt.Sprintf("error checking if %s index already exist: %s", indexName, err)
+		fmt.Println(reason)
+	} else if index.Shards.Successful < index.Shards.Total { // index already exist, but some shards are unavailable
+		reason := fmt.Sprintf("%s index already exist, but only %d shards are available out of %d", indexName,
+			index.Shards.Successful, index.Shards.Total)
+		fmt.Println(reason)
+		//	TODO: Should we raise a panic here?
+	} else {
+		reason := fmt.Sprintf("%s index already exist, so skipping creation", indexName)
+		fmt.Println(reason)
+		return
+	}
+	// as index is non-existent, so creating it
+	var result *elastic.IndicesCreateResult
+	var config1 Configuration
+	config1.Settings.Index.NumberOfShards = shardsCount
+	config1.Settings.Index.NumberOfReplicas = replicasCount
+
+	mappingbyte, err := json.Marshal(config1)
+	if err != nil {
+		fmt.Println("error in marshalling")
```

Comment:
Lets be more specific about the errors so that its easier to debug.
err = fmt.Errorf("error marshalling shard configuration for index %s | %s", indexName, err)
fmt.Println(err)
return